models/transaction_test.go

```diff
@@ -29,60 +56,208 @@ func createTestEvent(t *testing.T) (cadence.Event, *gethTypes.Transaction, *type
 		}},
 	}
 
-	tx := gethTypes.NewTx(&gethTypes.AccessListTx{
-		ChainID:  big.NewInt(1),
-		Nonce:    1,
-		GasPrice: big.NewInt(24),
-		Gas:      1337,
-		To:       &common.Address{0x01},
-		Value:    big.NewInt(5),
-		Data:     []byte{0x2, 0x3},
-	})
-
-	var txEnc bytes.Buffer
-	err := tx.EncodeRLP(&txEnc)
-	require.NoError(t, err)
-
 	ev := types.NewTransactionExecutedEvent(
 		1,
-		txEnc.Bytes(),
+		txEncoded,
 		common.HexToHash("0x1"),
-		tx.Hash(),
+		txHash,
 		res,
 	)
 
 	cdcEv, err := ev.Payload.CadenceEvent()
 	require.NoError(t, err)
 
-	return cdcEv, tx, res
+	return cdcEv, res
 }
 
-func Test_DecodeTransaction(t *testing.T) {
-	cdcEv, tx, _ := createTestEvent(t)
+func Test_DecodeEVMTransaction(t *testing.T) {
+	cdcEv, _ := createTestEvent(t, evmTxBinary)
 
 	decTx, err := DecodeTransaction(cdcEv)
 	require.NoError(t, err)
-	assert.Equal(t, tx.Hash(), decTx.Hash())
-	assert.Equal(t, tx.Type(), decTx.Type())
-	assert.Equal(t, tx.To(), decTx.To())
-	assert.Equal(t, tx.Value(), decTx.Value())
-	assert.Equal(t, tx.Nonce(), decTx.Nonce())
-	assert.Equal(t, tx.Data(), decTx.Data())
-	assert.Equal(t, tx.GasPrice(), decTx.GasPrice())
-	assert.Equal(t, tx.BlobGas(), decTx.BlobGas())
-	assert.Equal(t, tx.Size(), decTx.Size())
+	require.IsType(t, TransactionCall{}, decTx)
+
+	txHash, err := decTx.Hash()
+	require.NoError(t, err)
+	v, r, s := decTx.RawSignatureValues()
+	from, err := decTx.From()
+	require.NoError(t, err)
+
+	assert.Equal(
```

Comment:
For unit tests imho is better to not rely on fixture data. Generally it would be better to create the transaction that then gets encoded into the event (like it was before). If you think that is a lot of overhead for this tests maybe that's fine but I want to warn about this so it doesn't become a habit of just having fixtures data to do tests. Why I say this is because understanding what values are and should be is very hidden in the binary data in the fixture file. Also changing becomes a pain since the fixtures don't have a way to be generated by others.

Reply:
That's a fair point. I will refrain from using this type of input for other tests. But for this specific test, I wanted to use actual production input, so that we can see from their assertions the differences between the two types of state transitions.